pym/bob/cmds/build.py

```diff
@@ -948,15 +958,41 @@ def _cookPackageStep(self, packageStep, checkoutOnly, depth):
                         wasDownloaded = True
                     elif self.__forcedDownload:
                         raise BuildError("Downloading artifact failed")
-                elif oldWasDownloaded:
+                elif oldWasDownloadedOrShared:
                     self._info("   PACKAGE   skipped (already downloaded in {})".format(prettyPackagePath))
                     wasDownloaded = True
 
+            if packageStep.isShared() and self.__sharedFolder and not wasDownloaded:
```

Comment:
Isn't that already too late here? We would first try to download the artifact and then see if it's in the shared location...